android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -213,91 +385,435 @@ private void handleBodyCompositionMeasurement(byte[] value) {
+                }
+                if ((scaleUserList.size() == 0)) {
+                    storeUserScaleConsentCode(selectedUser.getId(), -1);
+                    storeUserScaleIndex(selectedUser.getId(), -1);
+                    jumpNextToStepNr(SM_STEPS.REGISTER_NEW_SCALE_USER.ordinal());
+                    resumeMachineState();
+                    return;
+                }
+                if (getUserScaleIndex(selectedUser.getId()) == -1 || getUserScaleConsent(selectedUser.getId()) == -1)  {
+                    chooseExistingScaleUser(scaleUserList);
+                    return;
+                }
+                resumeMachineState();
+                return;
+            }
+            int index = parser.getIntValue(FORMAT_UINT8);
+            String initials = parser.getStringValue();
+            int end = 3 > initials.length() ? initials.length() : 3;
+            initials = initials.substring(0, end);
+            if (initials.length() == 3) {
+                if (initials.charAt(0) == 0xff && initials.charAt(1) == 0xff && initials.charAt(2) == 0xff) {
+                    initials = "";
+                }
+            }
+            parser.setOffset(5);
+            int year = parser.getIntValue(FORMAT_UINT16);
+            int month = parser.getIntValue(FORMAT_UINT8);
+            int day = parser.getIntValue(FORMAT_UINT8);
+            int height = parser.getIntValue(FORMAT_UINT8);
+            int gender = parser.getIntValue(FORMAT_UINT8);
+            int activityLevel = parser.getIntValue(FORMAT_UINT8);
+            GregorianCalendar calendar = new GregorianCalendar(year, month - 1, day);
+            ScaleUser scaleUser = new ScaleUser();
+            scaleUser.setUserName(initials);
+            scaleUser.setBirthday(calendar.getTime());
+            scaleUser.setBodyHeight(height);
+            scaleUser.setGender(Converters.Gender.fromInt(gender));
+            scaleUser.setActivityLevel(Converters.ActivityLevel.fromInt(activityLevel - 1));
+            scaleUser.setId(index);
+            scaleUserList.add(scaleUser);
+            if (scaleUserList.size() == VENDOR_SPECIFIC_MAX_USERS) {
+                chooseExistingScaleUser(scaleUserList);
+            }
+    }
+
+    protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
+        final DateFormat dateFormat = DateFormat.getDateInstance();
+        int choicesCount = userList.size();
+        if (userList.size() < VENDOR_SPECIFIC_MAX_USERS) {
+            choicesCount = userList.size() + 1;
+        }
+        CharSequence[] choiceStrings = new String[choicesCount];
+        int indexArray[] = new int[choicesCount];
+        int selectedItem = -1;
+        for (int i = 0; i < userList.size(); ++i) {
+            ScaleUser u = userList.get(i);
+            String name = u.getUserName();
+            choiceStrings[i] = (name.length() > 0 ? name : String.format("P%02d", u.getId()))
```

Comment:
These strings are shown to the user right? If yes we might use resource strings for a translation support.

Reply:
Yes, agree, will substitute hard-coded string with resource strings.

Reply:
Fixed.